enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/grpc/ExtAuthService.java

```diff
@@ -151,7 +154,15 @@ private CheckResponse buildResponse(CheckRequest request, ResponseObject respons
                     .build();
         } else {
             OkHttpResponse.Builder okResponseBuilder = OkHttpResponse.newBuilder();
-
+            if (secProtocolHeaderForWS[0].equals(Constants.WS_API_KEY_IDENTIFIER) &&
+                secProtocolHeaderForWS.length == 2) {
+                okResponseBuilder.addResponseHeadersToAdd(
+                    HeaderValueOption.newBuilder()
+                        .setHeader(HeaderValue.newBuilder()
+                        .setKey(HttpConstants.WEBSOCKET_PROTOCOL_HEADER)
+                        .setValue(Constants.WS_API_KEY_IDENTIFIER).build())
+                        .build());
+            }
```

Comment:
I think it is better to move this logic to Internal Key Authenticator. We can add a new field for addResponseHeadersToAdd in the ResponseObject object and build OkResponse using that, the same way we are doing for adding request headers with responseObject.getHeaderMap()

Reply:
As this is related to API Key auth it is better to have it there